node/libs/storage/src/testonly/in_memory.rs

```diff
@@ -0,0 +1,71 @@
+//! In-memory storage implementation.
+use crate::{BlockStoreState, PersistentBlockStore, ReplicaState};
+use std::{collections::BTreeMap, sync::Mutex};
+use zksync_concurrency::ctx;
+use zksync_consensus_roles::validator;
+
+/// In-memory block store.
+#[derive(Debug, Default)]
+pub struct BlockStore(Mutex<BTreeMap<validator::BlockNumber, validator::FinalBlock>>);
```

Comment:
- Ditto as for `crate::BlockStore`; using a `VecDeque` could help clarifying an invariant that only continuous ranges of blocks can be stored.
- Bikeshedding: Subjectively, having multiple structs named `BlockStore` in the same crate is slightly confusing.

Reply:
* done
* ack, that's why we don't use fully qualified imports for these. We would need to prefix the struct names with the module name, which would cause unnecessary stuttering